tools/codegen/operator_versions/gen_mobile_upgraders.py

```diff
@@ -0,0 +1,285 @@
+
+def write_cpp(cpp_path, upgrader_dict):
+    body_parts = []
+
+    all_upgrader_src_string = []
+    for upgrader_bytecode in upgrader_dict:
+        for op_name, bytecode in upgrader_bytecode.items():
+            instruction_list_str = ""
+            constant_list_str = ""
+            type_list_str = ""
+            register_size_str = ""
+            operator_list_str = ""
+            for table_name, contents in bytecode.items():
+                # print("table name: ", table_name)
+                element = ByteCode[table_name]
+                body_string = ""
+                if element is ByteCode.instructions:
+                    instruction_list_str = construct_instruction(contents)
+                elif element is ByteCode.constants:
+                    constant_list_str = construct_constants(contents)
+                elif element is ByteCode.operators:
+                    operator_list_str = construct_operators(contents)
+                elif element is ByteCode.types:
+                    type_list_str = construct_types(contents)
+                elif element is ByteCode.register_size:
+                    register_size_str = construct_register_size(contents)
+
+            one_upgrader_function_string = ONE_UPGRADER_FUNCTION.substitute(
+                upgrader_name=op_name,
+                instruction_list=instruction_list_str,
+                constant_list=constant_list_str,
+                type_list=type_list_str,
+                register_size=register_size_str,
+            )
+            one_upgrader_src_string = ONE_UPGRADER_SRC.substitute(
+                bytecode_function=one_upgrader_function_string,
+                operator_string_list=operator_list_str,
+            )
+            all_upgrader_src_string.append(one_upgrader_src_string)
+
+    upgrader_file_content = UPGRADER_CPP_SRC.substitute(
+        operator_version_map="a",
+        upgrader_bytecode="".join(all_upgrader_src_string))
+    body_parts.append(upgrader_file_content)
+    print("writing file to : ", cpp_path + "/" + UPGRADER_MOBILE_FILE_NAME)
+    with open(
+        os.path.join(cpp_path, UPGRADER_MOBILE_FILE_NAME), "wb"
+    ) as out_file:
+        final_output = "".join(body_parts)
+        out_file.write(upgrader_file_content.encode("utf-8"))
+
+def parse_upgrader(upgrader_yaml_path: str) -> Dict:
+    upgrader_dict = load_yaml(upgrader_yaml_path)
+    return upgrader_dict
+
+def get_parser_options(parser: argparse.ArgumentParser) -> argparse.Namespace:
+    parser.add_argument(
+        "--input_upgrader_bytecode_yaml",
+        type=str,
+        help="A path to input yaml file.",
+        required=True,
+    )
+    parser.add_argument(
+        "--output_path",
+        help=f"Output {UPGRADER_MOBILE_FILE_NAME}.",
+        required=True,
+    )
+    options = parser.parse_args()
+    return options
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(description="Generate used operators YAML")
```

Comment:
Description seems to be for a different argument?

Reply:
Oh good call. Update in the latest commit.